rayon-core/src/registry.rs

```diff
@@ -611,6 +620,8 @@ pub(super) struct WorkerThread {
 // for a RefCell<T> etc.
 thread_local! {
     static WORKER_THREAD_STATE: Cell<*const WorkerThread> = Cell::new(ptr::null());
+
+    static LOCK_LATCH: RefCell<LockLatch> = RefCell::new(LockLatch::new());
```

Comment:
Why does this need a `RefCell`? From the outside it's only used immutably (no `borrow_mut`), and its inner mutex provides the only mutability we need.

Also, this thread-local doesn't need global scope -- let's move it into `worker_cold` where it's used. It will still be static, shared across multiple calls to that function.